### PR TITLE
[WIP] Log axes 2d

### PR DESCRIPTION
**Pull request by doutriaux1**
needs: https://github.com/UV-CDAT/vcs/pull/315

### CODE REVIEW

**Comment by doutriaux1** [comment]
@danlipsa @scottwittenburg I would like to merge this shortly in master, can you review please, it supersedes https://github.com/UV-CDAT/vcs/pull/315 
Baselines are at:
https://github.com/UV-CDAT/uvcdat-testdata/pull/188

**Comment by scottwittenburg** [comment]
Whew, huge change!  I mostly skimmed the parts I wasn't familiar with, but no potential issues popped out at me.

+1

**Comment by doutriaux1** [comment]
@danlipsa actually I did not do the vector or streamline graphic method. I don't quite follow the logic on how to propagate the  axes to the vector arrays. I'll double check streamlines, I think I just forgot about them but if they store their own vtk copy of the data then i think it will be up to you again.

**Comment by doutriaux1** [comment]
@danlipsa might be worth merging this in first and then fix vector/streamlines separately in another PR.

File: vcs/utils.py [hunk 1]
```diff
@@ -1856,6 +1856,30 @@ def prettifyAxisLabels(ticks, axis):
     return ticks
 
 
+axisConvertFunctions = {
+    "linear": {"forward": lambda x: x, "invert": lambda x: x},
+    "area_wt": {"forward": lambda x: numpy.sin(x / 180. * numpy.pi),
+                "invert": lambda x: numpy.arcsin(x) / numpy.pi * 180.},
```
Comment (danlipsa) [inline vcs/utils.py]: What is area_wt?

**Comment by danlipsa** [comment]
@doutriaux1 Should be the same as the rest of the 2D plots, isn't it.

**Comment by danlipsa** [comment]
Looks good to me. Nice work.

**Comment by danlipsa** [comment]
Sure we can merge this and do the vectors and streamlines in a different PR.

**Comment by doutriaux1** [comment]
@danlipsa make sure you review this first though, there's a lot going on here.

**Comment by doutriaux1** [comment]
@danlipsa when I apply to vectors (area_wt on y axis) I get tiny arrows.

**Comment by danlipsa** [comment]
@doutriaux1 I have a question on axisconvert: What is the purpose of area_wt. What does wt stand for?

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 @scottwittenburg You have 'xyscale' for both axes but 'xaxisconvert' and 'yaxisconvert' for individual axes. It would be nice to make them consistent.
What about xyaxisscale, xaxisscale, yaxisscale?

**Comment by danlipsa** [comment]
We could change only xaxisconvert and yaxisconvert to xscale and yscale

**Comment by doutriaux1** [comment]
@danlipsa `area_wt` weights (hence the `wt`) the latitudes by their area, essentually it's using `sine(lat)`.

**Comment by doutriaux1** [comment]
@danlipsa where do you see `xscale` and `yscale`?

**Comment by danlipsa** [comment]
I don't. This are the suggested new names for xaxisconvert and yaxisconvert so that they are consistent with xyscale.

**Comment by doutriaux1** [comment]
@danlipsa ok, I guess I'm wondering where do we use `xyscale` 😄

**Comment by danlipsa** [comment]
That's a minor issue. Grepping for xscale I can see we use that with a different meaning. Maybe:
xyaxisconvert (current name is xyscale)
xaxisconvert
yaxisconvert

**Comment by doutriaux1** [comment]
@danlipsa I see it now, there's indeed an `xyscale` function to set both xaxisconvert and yaxisconvert on the graphic method. I didn't even know about it! Feel free to rename the function or simply remove it, I don't it's ever used and it's really just bloatware in my opinion.

**Pull request merged**